### PR TITLE
[DistSQL] Support `count schema rules` syntax

### PR DESCRIPTION
**Pull request by lanchengx**
Examples are as follows
image

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    
+    private static final String DATASOURCE = "datasource";
+    
+    private static final String TABLE = "table";
+    
+    private static final Map<String, Class<? extends RuleConfiguration>> FEATURE_MAP = new HashMap<>(5);
+    
+    private Iterator<Collection<Object>> data;
+    
+    private final int defaultCount = 0;
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java:74]: Why not make it static?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/distsql/constant/ExportableConstants.java [hunk 2]
```diff
@@ -36,5 +36,7 @@
     
     public static final String REPLICA_DATA_SOURCE_NAMES = "replica_data_source_names";
     
+    public static final String SINGLE_TABLE_TABLES = "single_table_tables";
```
Comment (RaigorJiang) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/distsql/constant/ExportableConstants.java:39]: single_tables?

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    
+    private static final String BINDING_TABLE = "binding_table";
+    
+    private static final String BROADCAST_TABLE = "broadcast_table";
+    
+    private static final String DATASOURCE = "datasource";
+    
+    private static final String TABLE = "table";
+    
+    private static final Map<String, Class<? extends RuleConfiguration>> FEATURE_MAP = new HashMap<>(5);
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java:72]: loadFactor

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    
+    private static final String SHADOW = "shadow";
+    
+    private static final String SHARDING_TABLE = "sharding_table";
+    
+    private static final String BINDING_TABLE = "binding_table";
+    
+    private static final String BROADCAST_TABLE = "broadcast_table";
+    
+    private static final String DATASOURCE = "datasource";
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/rule/SchemaRulesQueryResultSet.java:68]: `datasource` is not a word, please change to  data_source

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**